platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationReport.java

```diff
@@ -0,0 +1,144 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * The migration report provides information about the execution of a migration operation.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface MigrationReport {
+    /**
+     * Gets the type of migration operation this report is associated with.
+     *
+     * @return the type of migration operation this report is associated with
+     */
+    public MigrationOperation getOperation();
+
+    /**
+     * Gets the time the corresponding migration operation started.
+     *
+     * @return the time from the epoch the report migration operation started
+     */
+    public long getStartTime();
+
+    /**
+     * Gets the time the corresponding migration operation ended.
+     *
+     * @return the time from the epoch the report migration operation ended or <code>-1L</code> if it
+     * is still in operation
+     */
+    public long getEndTime();
+
+    /**
+     * Records a warning that occurred during the migration report.
+     *
+     * @param w the warning to record
+     * @return this for chaining
+     * @throws IllegalArgumentException if <code>w</code> is <code>null</code>
+     */
+    public MigrationReport record(MigrationWarning w);
+
+    /**
+     * Records an error that occurred during the migration report.
+     * <p>
+     * Recorded errors will be thrown back when {@link #verifyCompletion()} is called at the end of
+     * the operation.
+     *
+     * @param e the error to record
+     * @return this for chaining
+     * @throws IllegalArgumentException if <code>e</code> is <code>null</code>
+     */
+    public MigrationReport record(MigrationException e);
+
+    /**
+     * Registers code to be invoked at the completion of the migration operation.
+     *
+     * @param code the code to execute which will receive this report in parameter where
+     *             additional errors and/or warnings can be registered
+     * @return this for chaining
+     * @throws IllegalArgumentException if <code>code</code> is <code>null</code>
+     */
+    public MigrationReport doAfterCompletion(Consumer<MigrationReport> code);
```

Comment:
Definitely not in the migratable since the migratable is implemented by implementors and this is a functionality provided by the framework implementation for use by the migratables.